tests/integration_tests/build/test_coverage.py

```diff
@@ -32,6 +32,43 @@
 """Regex for extracting number of total executable lines found by kcov."""
 
 
+@pytest.mark.timeout(120)
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
```

Comment:
nit: I understand that we can assume that this test could run once per CI run. However, if one developer uses an arm machine for development and this test fails, he would find out only after the CI finished running the tests. The issue with this is that it will add a delay in getting the PR in a "green" state, assuming that the development on one architecture must be compliant with the tests on any other architecture and the PR author can not test on all architectures, before submitting the PR. I can not tell if this is a very pressing issue, because I do not know how many developers are pushing code to firecracker from an arm machine exclusively and we should take this issue into consideration only if it is a real issue. I just wanted it to call it out, because it seems to that this can be easily fixed if we would have some kind of CI part which is common for all tested platforms and must run once. I do not know if we should think about this too much right now or take it into consideration, but I am curious how you guys feels about it.

Reply:
If a developer were developing on arm, then because kcov never runs on arm, the coverage report would always be unknown,  and they'd have to push and trigger a CI run to get the coverage percentage anyway (maybe? Or run it on x86_64 to get the figure).  At this point they'd find out if they'd misnamed anything.  

It feels like this test and the kcov run should be linked via architecture, but I don't mind changing it if anyone feels strongly about it.

Reply:
Hmm, I suggested to change the reason there to also not link 100% with `kcov` because also for the uniformity / consistency reason I prefer to have all unit tests in `mod tests`.
Anyway, @iulianbarbu has raised a valid issue here, at this moment we are skipping quite a few tests on arm (mostly style tests). Maybe we should add an issue for that? In the meantime this PR can be unblocked I guess.

Reply:
> If a developer were developing on arm, then because kcov never runs on arm, the coverage report would always be unknown, and they'd have to push and trigger a CI run to get the coverage percentage anyway (maybe? Or run it on x86_64 to get the figure). At this point they'd find out if they'd misnamed anything.
> 
> It feels like this test and the kcov run should be linked via architecture, but I don't mind changing it if anyone feels strongly about it.

Yes, at the moment, because of this test and kcov on arm, it seems that makes sense to have some tests run via architecture. `kcov` on arm has some instability issues. I do not know if this is the reason, or there are others, but because of this lack of 100% support on arm, we are coerced to enforce coverege test to run on `x86_64` arch only.

The reason I was proposing the platform independent execution CI part, which executes once all the tests that are platform independent, is because this test and others (style test), does some static analysis on the source code. If there would be platform dependent results, than it will make sense to make the test platform aware. In our current CI infrastructure, to run this test as if it is a platform independent test, once, we would need the `platform.machine()` check to skip it if the platform is not `x86_64`. However, this will not be consistent if the CI is run only on an arm machine, as well as if a developer runs the integration tests on an arm machine only.

To change this means to change the style checks as well. My thought is that we can stick to this solution at the moment and track this suggestion of platform independent tests a issue. I will open an issue for this, and link it to this PR.

Reply:
Yes, sorry, you're absolutely right. :-)